sql/api/src/main/scala/org/apache/spark/sql/internal/types/AbstractStringType.scala

```diff
@@ -46,9 +46,10 @@ case object StringTypeBinaryLcase extends AbstractStringType {
 }
 
 /**
- * Use StringTypeAnyCollation for expressions supporting all possible collation types.
+ * Use StringTypeWithCaseAccentSensitivity for expressions supporting all collation types
+ * (binary and ICU) but limited to using case and accent sensitivity specifiers.
  */
-case object StringTypeAnyCollation extends AbstractStringType {
+case object StringTypeWithCaseAccentSensitivity extends AbstractStringType {
```

Comment:
What about pass-through expressions? Shouldn't we keep both names and use `StringTypeWithCaseAccentSensitivity` for those expressions that we don't have trim support yet and `StringTypeAnyCollation` for pass-through expressions (as they won't be affected in the future)?

Reply:
StringTypeAnyCollation is name that we shouldn't use for good. If collations supports use of all specifiers, just embed it in the name to avoid any confusion.